packages/nodes-base/nodes/Hubspot/Hubspot.node.ts

```diff
@@ -935,8 +1026,206 @@ export class Hubspot implements INodeType {
+
+			const batches = Math.ceil(length / maxBatchSize);
+			const resultsPromises = [...new Array(batches)].map(async (_, batchNumber) => {
+				try {
+					const batchStart = batchNumber * maxBatchSize;
+					const batchSize = Math.min(length - batchStart, maxBatchSize);
+					const batchEnd = batchStart + batchSize;
+
+					if (operation === 'batchGet') {
+						const additionalFields = this.getNodeParameter('additionalFields', 0) as IDataObject;
+						const idProperty = this.getNodeParameter('idProperty', 0) as string | null;
+
+						const requestBody = {
+							properties: additionalFields.properties as string[] || [],
+							propertiesWithHistory: additionalFields.propertiesWithHistory as string[] || [],
+							idProperty: idProperty || undefined,
+							inputs: ([...new Array(batchSize)]).map((_, index) => ({ id: this.getNodeParameter('objectId', index + batchStart) as string })),
+						};
+						const endpoint = `/crm/v3/objects/${customObjectType}/batch/read`;
+						const response = await hubspotApiRequest.call(this, 'POST', endpoint, requestBody);
+
+						const results = response.results;
+
+						results.push(...getErrorsFromBatchResponse(response));
+
+						return results;
+					}
+					if (operation === 'batchDelete') {
+						const idProperty = this.getNodeParameter('idProperty', 0) as string | null;
+						const errors: Array<{ error: string, errorDetails: unknown }> = [];
+
+						// Resolve object ids, if a custom idPorperty is used
+						const idMap: Record<string, string | undefined> = {};
+						if (idProperty) {
+							const requestBody = {
+								idProperty,
+								inputs: ([...new Array(batchSize)]).map((_, index) => ({ id: this.getNodeParameter('objectId', index + batchStart) as string })),
+							};
+							const endpoint = `/crm/v3/objects/${customObjectType}/batch/read`;
+							const response = await hubspotApiRequest.call(this, 'POST', endpoint, requestBody);
+							response.results.forEach((result: { id: string, properties: Record<string, string> }) => {
+								idMap[result.properties[idProperty]] = result.id;
+							});
+						} else {
+							([...new Array(batchSize)]).forEach((_, index) => {
+								const id = this.getNodeParameter('objectId', index + batchStart) as string;
+								idMap[id] = id;
+							});
+						}
+
+						const idsToDelete = [...new Array(batchSize)].map((_, index) =>
+							idMap[this.getNodeParameter('objectId', index + batchStart) as string],
+						).filter(id => id !== undefined) as string[];
+
+						const requestBody = {
+							inputs: idsToDelete.map(id => ({
+								id,
+							})),
+						};
+						const endpoint = `/crm/v3/objects/${customObjectType}/batch/archive`;
+						const response = await hubspotApiRequest.call(this, 'POST', endpoint, requestBody);
+						errors.push(...(response ? getErrorsFromBatchResponse(response) : []));
+
+						const results = response?.results || [];
```

Comment:
result is always empty

Reply:
Hello @Zebreus , thank you for your contribution, I did review and I'm not sure why batch operations needs to be dependent on continue on fail, I do not see any reason they should be a case but maybe I'm missing something. In any way we need to find a way to not do this. There are some other problems with upsert and delete operations, please take a look at comments above. Also some properties that depends on load methods need to updated with loadOptionsDependsOn.
If you have any questions please let me know.